compose/foundation/foundation/src/desktopMain/kotlin/androidx/compose/foundation/internal/ClipboardUtils.desktop.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.foundation.internal
+
+import androidx.compose.ui.platform.ClipEntry
+import androidx.compose.ui.text.AnnotatedString
+import java.awt.datatransfer.DataFlavor
+import java.awt.datatransfer.StringSelection
+import java.awt.datatransfer.UnsupportedFlavorException
+import java.io.IOException
+
+internal actual suspend fun ClipEntry.readText(): String? {
+    return try {
+        transferable.getTransferData(DataFlavor.stringFlavor) as String?
```

Comment:
I think `getTransferData` can be a long-running operation (if there's a lot of text, for example), and IDEA rightfully complains about "Possibly blocking call in non-blocking context could lead to thread starvation".

We should wrap the call with `withContext(Dispatchers.IO){ ... }` to avoid blocking the main thread.

Reply:
I don't actually know how important this is, but it's possible that we need to handle more than just `stringFlavor`. Take a look at `Transferable.dragData()` in `AwtDragData.kt`.

Reply:
> We should wrap the call with withContext(Dispatchers.IO){ ... } to avoid blocking the main thread.

The previous implementation (from skiko) was always blocking. It didn't cause any issues, perhaps due to API being used rather rarely.

And the suspend functions were added only for web. 
I think adding `withContext` won't hurt. 

____
> but it's possible that we need to handle more than just stringFlavor

For the internal needs, they are not needed right now (I mean the internal implementations of the features relying on Clipboard). But in general - yes, after an API design and proposal to Jetpack Compose.